src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -12419,6 +12471,471 @@ public static void Main()
+                //         Del del = M;
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "M").WithArguments("2", @"""b""", @"""a""").WithLocation(8, 19),
+                // (8,19): warning CS9069: Parameter 3 has default value '1' in method group and 'default(long)' in the target delegate type.
+                //         Del del = M;
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "M").WithArguments("3", "1", "default(long)").WithLocation(8, 19));
+
+        }
+
+        [Fact]
+        public void ExtensionMethodGroup_DefaultParameters_TargetTypeConversion()
+        {
+            var source = """
+using System;
+using ProgramExtensions;
+
+namespace ProgramExtensions {
+    public static class ProgramExtensions
+    {
+        public static string M(this Program p, string s = "b", long l = 1L) => $"{p.field} {s} {l}";
+    }
+}
+
+public class Program
+{
+    public int field = 10;
+    delegate string Del(string s = "a", long l = 0L);
+    public string M1(string s = "c", long l = 2L) => $"{this.field} {s} {l}";
+
+    public static void Main()
+    {
+        Program prog = new Program();
+        Del del = prog.M;
+        Console.WriteLine(del());
+    }
+}
+""";
+            var verifier = CompileAndVerify(source, verify: Verification.FailsILVerify);
+            verifier.VerifyDiagnostics(
+                // (20,19): warning CS9068: Parameter 2 has default value '"b"' in method group and '"a"' in the target delegate type.
+                //         Del del = prog.M;
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "prog.M").WithArguments("2", @"""b""", @"""a""").WithLocation(20, 19),
+                // (20,19): warning CS9068: Parameter 3 has default value '1' in method group and 'default(long)' in the target delegate type.
+                //         Del del = prog.M;
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "prog.M").WithArguments("3", "1", "default(long)").WithLocation(20, 19));
+            verifier.VerifyIL("Program.Main",
+@"
+ {
+  // Code size       34 (0x22)
+  .maxstack  3
+  IL_0000:  newobj     ""Program..ctor()""
+  IL_0005:  ldftn      ""string ProgramExtensions.ProgramExtensions.M(Program, string, long)""
+  IL_000b:  newobj     ""Program.Del..ctor(object, System.IntPtr)""
+  IL_0010:  ldstr      ""a""
+  IL_0015:  ldc.i4.0
+  IL_0016:  conv.i8
+  IL_0017:  callvirt   ""string Program.Del.Invoke(string, long)""
+  IL_001c:  call       ""void System.Console.WriteLine(string)""
+  IL_0021:  ret
+}");
+        }
+
+        [Fact]
+        public void ExtensionMethodGroup_DefaultParameters_TargetTypeConversion_02()
+        {
+            var source = """
+namespace ProgramExtensions {
+    public static class PExt
+    {
+        public static string M(this Program p, string s = "b", long l = 1) => $"{p.field} {s} {l}";
```

Comment:
>string s = "b", long l = 1 [](http://example.com/codeflow?start=47&length=26)

It looks like the overload resolution error is unrelated to default values. Consider using the same default values as the delegate so that is more obvious.